pkg/apis/beat/v1beta1/validations_test.go

```diff
@@ -207,3 +209,126 @@ func Test_checkNoDowngrade(t *testing.T) {
 		})
 	}
 }
+
+func Test_checkMonitoring(t *testing.T) {
+	tests := []struct {
+		name string
+		beat *Beat
+		want field.ErrorList
+	}{
+		{
+			name: "stack monitoring not enabled returns nil",
+			beat: &Beat{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testbeat",
+					Namespace: "test",
+				},
+				Spec: BeatSpec{
+					Type:      "filebeat",
+					Version:   "8.2.3",
+					DaemonSet: &DaemonSetSpec{},
+				},
+			},
+			want: nil,
+		},
+		{
+			name: "too old version with stack monitoring enabled returns unsupported version error",
+			beat: &Beat{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testbeat",
+					Namespace: "test",
+				},
+				Spec: BeatSpec{
+					Type:      "filebeat",
+					Version:   "7.1.0",
+					DaemonSet: &DaemonSetSpec{},
+					Monitoring: Monitoring{
+						Metrics: MetricsMonitoring{
+							ElasticsearchRefs: []commonv1.ObjectSelector{
+								{
+									Name:      "elasticsearch",
+									Namespace: "test",
+								},
+							},
+						},
+					},
+				},
+			},
+			want: field.ErrorList{&field.Error{Type: field.ErrorTypeInvalid, Field: "spec.version", BadValue: "7.1.0", Detail: "Unsupported version for Stack Monitoring. Required >= 7.2.0."}},
```

Comment:
Would it make sense to have a test case that covers invalid version + invalid ref which as you mentioned is not properly covered in the common validation code?

Reply:
Invalid version isn't relevant since moving to sidecars; removed.